redhat-marketplace/redhat-marketplace-operator
# Bugfix/ocp414

workflow
- get the kubebuilder assets for testing from the standard make target: envtest
- move operator test down to test matrix

Makefile
- envtest used for get & testing with kubebuilder assets

airgap
- mod update to k8s v0.27.6, cotroller-runtime v0.15.2 for OpenShift 4.14
- generate

authchecker
- mod update to k8s v0.27.6, cotroller-runtime v0.15.2 for OpenShift 4.14

datareporter
- mod update to k8s v0.27.6, cotroller-runtime v0.15.2 for OpenShift 4.14
- update controller for api changes
- setup test suite (no working tests yet)

deployer
- mod update to k8s v0.27.6, cotroller-runtime v0.15.2 for OpenShift 4.14
- update controllers for api changes, injection option no longer exists
- rewrite tests removing rectest for standard controller-runtime CRUD
- check razee name in predicate instead of in reconcile, resolves occasional `bad name` test failure

metering
- mod update to k8s v0.27.6, cotroller-runtime v0.15.2 for OpenShift 4.14
- fix and re-enable tests
- update mocks for api changes

reporter
- mod update to k8s v0.27.6, cotroller-runtime v0.15.2 for OpenShift 4.14
- remove use of reconcileutils, rectest
- rewrite tests removing rectest for standard controller-runtime CRUD
- fix test result data for prometheus

v2 (rhm-operator)
- mod update to k8s v0.27.6, cotroller-runtime v0.15.2, prometheus, for OpenShift 4.14
- update controllers for api changes, injection option no longer exists
- remove use of reconcileutils, rectest
- rewrite tests removing rectest for standard controller-runtime CRUD
- remove reconcileutils, rectest
- remove tests/v2
  - controller suite is used, they were no longer run
- remove skaffold-tdd-tool
  - unused, no longer maintained
